app/controllers/ops_controller/settings/analysis_profiles.rb

```diff
@@ -239,10 +239,13 @@ def ap_edit
         return unless load_edit("ap_edit__#{id}", "replace_cell__explorer")
         @scan = ScanItemSet.find_by_id(@edit[:scan_id])
         ap_get_form_vars
-        if (@edit[:new]["category"].nil? || @edit[:new]["category"][:definition]["content"].length == 0) &&
-           (@edit[:new]["file"].nil? || @edit[:new]["file"][:definition]["stats"].length == 0) &&
-           (@edit[:new]["registry"].nil? || @edit[:new]["registry"][:definition]["content"].length == 0) &&
-           (@edit[:new]["nteventlog"].nil? || @edit[:new]["nteventlog"][:definition].nil? || @edit[:new]["nteventlog"][:definition]["content"].length == 0)
+
+        category, nteventlog, registry = %w(category nteventlog registry).map do |x|
+          @edit[:new].fetch_path(x, :definition, 'content')
+        end
+        file = @edit[:new].fetch_path('file', :definition, 'stats')
+
+        if [category, nteventlog, registry, file].all? { |item| item.nil? || item.length.zero? }
```

Comment:
imho `item.nil? || item.length.zero?` can be replaced with `item.blank?`

Reply:
```
irb(main):001:0> 0.blank?
=> false
irb(main):002:0> 0.zero?
=> true
```

Reply:
That is correct. But you have `item.length.zero?`

Reply:
```
2.5.0 :001 > [].blank?
 => true 
2.5.0 :002 > nil.blank?
 => true 
```

Reply:
Yes, you're right, thanks. Fixed.